crates/pathfinder/src/ethereum/log/fetch/forward.rs

```diff
@@ -38,22 +38,30 @@ impl<T> LogFetcher<T>
 where
     T: MetaLog + PartialEq + std::fmt::Debug + Clone,
 {
-    /// Creates a [LogFetcher] which fetches logs starting from `last_known`'s origin on L1.
-    /// If `last_known` is [None] then the starting point is genesis.
+    /// Creates a [LogFetcher] which fetches logs starting from `head`'s origin on L1.
+    /// If `head` is [None] then the starting point is genesis.
     ///
-    /// In other words, the first log returned will be the one after `last_known`.
-    pub fn new(last_known: Option<T>, chain: Chain) -> Self {
+    /// In other words, the first log returned will be the one after `head`.
+    pub fn new(head: Option<T>, chain: Chain) -> Self {
         let base_filter = FilterBuilder::default()
             .address(vec![T::contract_address(chain)])
             .topics(Some(vec![T::signature()]), None, None, None);
 
         Self {
-            last_known,
+            head,
             stride: 10_000,
             base_filter,
         }
     }
 
+    pub fn set_head(&mut self, head: Option<T>) {
+        self.head = head;
+    }
+
+    pub fn head(&self) -> &Option<T> {
+        &self.head
+    }
```

Comment:
Dunno if clippy is saying anything on this, but in general idiomaticness `Option<&T>` as in `self.head.as_ref()` is more fun. It's as powerful as the `&Option<T>` for the caller, but you can just consume it.